src/lib/Exporter/Exporter.php

```diff
@@ -129,25 +136,54 @@ private function getContentForGivenLanguages(
     private function generateFileForContentType(
         int $contentTypeId,
         string $chunkPath,
+        string $language,
+        int $page,
         Parameters $parameters,
-        OutputInterface $output): void
-    {
-        $content = $this->repository->sudo(
+        OutputInterface $output
+    ): void {
+        $contentList = $this->repository->sudo(
             function () use ($contentTypeId, $parameters, $output) {
                 return $this->contentService->fetchContent($contentTypeId, $parameters, $output);
             }
         );
 
+        $fileSettings = new FileSettings(
+            $this->getItems($contentList),
+            (string) $contentTypeId,
+            $language,
+            $page,
+            $chunkPath
+        );
+
         $output->writeln(sprintf(
             'Generating file for contentTypeId: %s, language: %s, chunk: #%s',
             $contentTypeId,
-            $parameters->lang,
-            $parameters->page
+            $language,
+            $page,
         ));
 
-        $this->exportFileGenerator->generateFile($content, $chunkPath, $parameters->getProperties());
+        $this->exportFileGenerator->generate($fileSettings);
+
+        unset($contentList);
+    }
+
+    /**
+     * @param iterable<\eZ\Publish\API\Repository\Values\Content\Content> $contentList
+     */
+    private function getItems(iterable $contentList): ItemListInterface
+    {
+        $itemList = [];
+        /** @var \eZ\Publish\API\Repository\Values\Content\Content $content */
```

Comment:
Provide separate information for PHPStan and PhpStorm, so neither of the tools get confused.

This achieves:
1. You can have phpstorm autocompletion.
2. You can have your types checked properly by PHPStan.
3. Type declarations are kept in one place.

```suggestion
     * @param \eZ\Publish\API\Repository\Values\Content\Content[] $contentList
     * @phpstan-param iterable<\eZ\Publish\API\Repository\Values\Content\Content> $contentList
     */
    private function getItems(iterable $contentList): ItemListInterface
    {
        $itemList = [];
```